_includes/manuals/nightly/3.7_migration.md

```diff
@@ -0,0 +1,171 @@
+called `foreman-maintain` when installed to ease the process of shutting down
+services. Make sure that neither Puppet nor a monitoring system automatically
+starts these services again.
+
+You can issue the MySQL command `SHOW FULL PROCESSLIST;` to show you all active
+connections to your MySQL database. Make sure that there are none.
+
+The foreman user that has access to your Postgresql database and is used for
+the conversion temporarily needs super user privileges.
+
+```bash
+## postgres@postgres:~  $ su - postgres
+## postgres@postgres:~  $ psql
+psql (9.6.2)
+Type "help" for help.
+
+postgres=# ALTER USER foreman WITH SUPERUSER;
+ALTER ROLE
+postgres=#
+```
+
+#### Pre-migrate cleanup (optional)
+
+Before proceeding migration, it's a good idea to permanently delete all audit
+records older than few days, unless they are required for compliance reasons:
+
+    foreman-rake audits:expire days=10 RAILS_ENV="production"
+
+To permanently delete all reports older than 2 days:
+
+    foreman-rake reports:expire days=2 batch_size=50000 sleep_time=0.000001 RAILS_ENV="production"
+
+The command above deletes records in batches of 50k items, it can take a lot of
+time to process all records. In both cases, progress can be seen in the
+`production.log`.
+
+#### Migrating
+
+To migrate the data from one database to the other, we will use a rake task.
+We will be using `foreman-rake db:convert:prod2dev` to migrate the data.
+
+Foreman internally uses a development framework called Rails. In Rails, you can
+have a specific configuration for e.g. a development and a production
+environment. Our prod2dev rake task assumes, that the source database is
+configured as the production database and the new target database is configured
+as the development database. All data in the development database will be
+overwritten.
+
+You can configure the new database as Foreman's new development database by
+adding a new connection in `/etc/foreman/database.yml`.
+
+It may look like this:
+
+```yaml
+# The new target database
+development:
+  adapter: postgresql
+  host: postgres.example.com
+  port: 5432
+  database: foreman
+  username: foreman
+  password: "foreman"
+  pool: 25
+
+# The old source database
+production:
+  adapter: mysql2
+  host: mysql.example.com
+  port: 3306
+  database: foreman
+  username: foreman
+  password: "foreman"
+  pool: 25
+```
+
+In the next step, call a rake task which will create new database:
+
+```sh
+forman-rake db:create RAILS_ENV=development
+```
+
+The new database needs to be set up with all the tables required for Foreman to
+operate, so in order to prepare the new database to receive the actual data, we
+need to run all of Foreman's database migrations.
+
+```sh
+foreman-rake db:migrate RAILS_ENV=development
+```
+
+Before proceeding with the actual data migration, it's required to do separate
+migration of dynflow tables.
+
+```sh
+foreman-rake dynflow:migrate
```

Comment:
Isn't this step done as part of the migration script now? Thinking of which, should we also run the regular migration task as part of it? (my main though is this is a very long process, the fewer manual steps the user has to do the better)

Reply:
I don't think so. More than that, it's not yet merged: https://github.com/theforeman/foreman/pull/6962

Reply:
That will be merged once @adamruzicka gets back to the comments. 
My thought is that these tasks should be idempotent, so we can call them as part of the prod2dev flow to make sure the target db is in the right state. if we do that, we can make these instruction simpler for the users - so less chance of missing a step causing the next step to fail.